src/java.base/unix/native/libjava/ProcessImpl_md.c

```diff
@@ -559,9 +559,16 @@ spawnChild(JNIEnv *env, jobject process, ChildStuff *c, const char *helperpath)
     }
     offset = copystrings(buf, 0, &c->argv[0]);
     offset = copystrings(buf, offset, &c->envv[0]);
-    if (sp.dirlen > 0 && c->pdir != NULL) {
-        memcpy(buf+offset, c->pdir, sp.dirlen);
-        offset += sp.dirlen;
+    if (c->pdir != NULL) {
+        if (sp.dirlen > 0) {
```

Comment:
As long as c->pdir is non-null, I think the code below is safe to execute. `memcpy(a, b, len)` should be okay if `len` is 0, as long as `a` and `b` are non-null, right?

So this check here is not needed, I think.